### ISSUE TITLE
Validation of REST input/output classes defined as dependency

### ISSUE THREAD

**Issue by mferlan** [quarkusio/quarkus#9598]
**Describe the bug**
As a developer I use Hibernate Validator/Bean Validation for validating the input/output of your REST services. 
But input/output of my REST services is defined in separate library (generated from RAML or OpenAPI file).

**Expected behavior**
When I follow **validation-quickstart** guide and make modification to extract my model to dependency then hibernate-validator should validate input/output of my REST services

**Actual behavior**
Response code 400 is returned with following message
`javax.validation.UnexpectedTypeException: HV000030: No validator could be found for constraint 'javax.validation.constraints.Size' validating type 'java.lang.String'. Check configuration for 'validateBook.arg0.author'`

**To Reproduce**
1. Follow **[validation-quickstart](https://quarkus.io/guides/validation)** guide.
2. Afterwards, a separate project `validation-quickstart-model `project is created
   - class Book is moved to that project.
   - added dependency to  `javax.validation:validation-api:jar:2.0.1.Final` with scope `provided`
   - validation-quickstart-model installed in local repository
3. Then in validation-quickstart project 
    -  added dependency to `validation-quickstart-model`
   -  added to application.properties following
```
quarkus.index-dependency.validation-quickstart.group-id=org.acme
quarkus.index-dependency.validation-quickstart.artifact-id=validation-quickstart-model
```
4. Open [http://localhost:8080/](http://localhost:8080/)
5. Enter 1 for number of pages and press 

Same behaviour  occurs when instead of adding quarkus.index-dependency to application.properties we build `validation-quickstart-model` with  `jandex-maven-plugin:1.0.7` plugin ( using goal `jandex`).

The only case in which it did work was by creating empty beans.xml in `validation-quickstart-model` project. 

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
`Linux mferlan-lap.at.local 3.10.0-1062.4.1.el7.x86_64 #1 SMP Fri Oct 18 17:15:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux` 
- Output of `java -version`: 
`java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)`
 - Quarkus version or git rev: 
**1.4.2.Final**
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/ferlan/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 1.8.0_101, vendor: Oracle Corporation, runtime: /opt/jdk1.8.0_101/jre
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-1062.4.1.el7.x86_64", arch: "amd64", family: "unix"
```
*** added on edit  ***
Proposal for using
 - dependencies that contain a beans.xml descriptor (content is ignored),
 -  dependencies that contain a Jandex index - META-INF/jandex.idx,
- dependencies referenced by quarkus.index-dependency in application.properties,

was based on [Context and Dependency Injection](https://quarkus.io/guides/cdi-reference).

**Comment by geoand** [comment]
Thanks for reporting!

Can you perhaps attach a project where you have already performed steps 1-3?

**Comment by mferlan** [comment]
[quarkus-validation.zip](https://github.com/quarkusio/quarkus/files/4680894/quarkus-validation.zip)
two project created following steps 1-3
Additionally, 
 -  pom.xml contains commented out configuration for jandex-maven-plugin
- there is an empty file src/main/resources/META-INF/empty-beans.xml 
   (when this file is renamed to beans.xml then validation works)

**Comment by gsmet** [comment]
> there is an empty file src/main/resources/META-INF/empty-beans.xml
(when this file is renamed to beans.xml then validation works)

Yes, that's because you need a `beans.xml` in your additional module. It needs to be a bean archive for the constraint annotations to be found.

**Comment by gsmet** [comment]
I thought I added something in the doc somewhere but I can't find it. Will have a closer look and if not there, I will add it.

**Comment by geoand** [comment]
> * there is an empty file src/main/resources/META-INF/empty-beans.xml
>   (when this file is renamed to beans.xml then validation works)

I'm very interested in knowing how you came up with `empty-beans.xml`. Was it something in the documentation that made you think this?

**Comment by mferlan** [comment]
> > * there is an empty file src/main/resources/META-INF/empty-beans.xml
> >   (when this file is renamed to beans.xml then validation works)
> 
> I'm very interested in knowing how you came up with `empty-beans.xml`. Was it something in the documentation that made you think this?

The intent regarding empty-beans.xml file remark is to point out that my attached project contained some "commented out solutions" and one of them was using empty beans.xml. So instead of deleting and recreating a file, for me it is far easier to rename a file and test if it works.

**Regarding documentation**
When I referenced model as dependency, then Quarkus build failed and one of the suggestion was to reference classes by quarkus.index-dependency in application.properties.
This resolved my build, but hibernate-validation  failed.

So after searching in guides for some clue, I ended up reading [Context and Dependency Injection Guide](https://quarkus.io/guides/cdi-reference). 
So, next best solution seemed trying adding jandex-maven-plugin to my "model" project.
This failed, and then I started writing this bug and wanted to test on a clean project and tested empty bean.xml and it worked.

I am fortunate that I can modify my "model jar" and add empty beans.xml.

I am not sure if this is a bug or not.
But it seemed useful feature to have possibility to validate "model" from third party that does not have beans.xml (that is not bean archive).
